app/policies/admin_grove_playlist_policy.rb

```diff
@@ -0,0 +1,5 @@
+class AdminGrovePlaylistPolicy < ApplicationPolicy
+  def show?
+    user.present? && user.is_a?(Teacher)
```

Comment:
might be OK here to do `user.admin?` and implement that method on the student to return false. use duck typing to our advantage there.